joern-cli/frontends/javasrc2cpg/src/main/scala/io/joern/javasrc2cpg/passes/TypeInfoProvider.scala

```diff
@@ -323,6 +323,22 @@ class TypeInfoProvider(global: Global) {
       }
     }
   }
+
+  def scopeType(scopeContext: ScopeContext, isSuper: Boolean = false): String = {
+    scopeContext.typeDecl match {
+      case Some(typ) =>
+        if (isSuper) {
+          val parentType =
+            typ.inheritsFromTypeFullName.headOption
+              .getOrElse("ANY")
+          registerType(parentType)
+        } else {
+          registerType(typ.fullName)
+        }
+
+      case None => "ANY"
```

Comment:
```suggestion
      case Some(typ) if isSuper =>
        val parentType = typ.inheritsFromTypeFullName.headOption.getOrElse("ANY")
        registerType(parentType)
      case Some(typ) =>
        registerType(typ.fullName)
      case None => "ANY"
```

matter of personal style I guess, so feel free to disregard